packages/office-ui-fabric-react/src/components/Button/ButtonBase.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import {
+  BaseComponent,
+  css,
+  assign,
+  getId,
+  getNativeProps,
+  buttonProperties,
+  anchorProperties
+} from '../../Utilities';
+import { IButtonProps, IButton, ButtonType } from './Button.Props';
+
+export interface IButtonState {
+  labelId?: string;
+  descriptionId?: string;
+  ariaDescriptionId?: string;
+}
+
+export class ButtonBase extends BaseComponent<IButtonProps, IButtonState> implements IButton {
+  public static defaultProps: IButtonProps = {
+    buttonType: ButtonType.default
+  };
+  private _buttonElement: HTMLButtonElement;
+
+  constructor(props: IButtonProps) {
+    super(props, { 'rootProps': null });
+
+    this.state = {
+      labelId: getId('Button'),
+      descriptionId: getId('Button'),
+      ariaDescriptionId: getId('Button')
+    };
+  }
+
+  public render(): JSX.Element {
+    let { buttonType, children, icon, description, ariaLabel, ariaDescription, href, disabled, onClick } = this.props;
+    let { labelId, descriptionId, ariaDescriptionId } = this.state;
+
+    const renderAsAnchor: boolean = !!href;
+    const tag = renderAsAnchor ? 'a' : 'button';
+    const nativeProps = getNativeProps(this.props.rootProps || this.props, renderAsAnchor ? anchorProperties : buttonProperties);
+    const className = css((this.props.className));
+
+    const iconSpan = icon && (buttonType === ButtonType.command || buttonType === ButtonType.hero || buttonType === ButtonType.icon)
```

Comment:
So we are okay with no backwards compatibility? Old <button>s will need to be updated or not render?